core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2432,6 +2432,18 @@ private[spark] object Utils extends Logging {
   }
 }
 
+private[util] object CallerContext {
+  val callerContextSupported: Boolean = {
+    SparkHadoopUtil
```

Comment:
Let's find a different way to wrap this; it's non-standard and hard to read the condition here. Maybe break out `conf` as a local `val` to make it simpler.

I see why you're using Try, though I wonder if we are suppressing potentially important errors this way. For example, ClassNotFoundException obviously means 'not supported' and that's normal. A fatal error should probably just propagate. But a NonFatal error could be logged with a warning? because that would be unexpected. It would still mean the caller context isn't supported rather than cause a failure.